functors_task_1/student/include/impl.h

```diff
@@ -40,9 +80,17 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink D1, NonAlcoholDrink D2)
     {
+        static const std::map<std::pair<AlcoholDrink, NonAlcoholDrink>, Cocktail> Cookbook = {
```

Comment:
Nice solution, but this task we should resolve statically, without containers usage.